tests/test_z_numpy.py

```diff
@@ -14,3 +17,57 @@ def test_z_numpy_ndarray_is_tensorflow_tensor():
     assert znp.ndarray is tf.Tensor
     assert isinstance(znp.array(1), tf.Tensor)
     assert isinstance(znp.sum(znp.array(0)), tf.Tensor)
+
+
+def test_faddeeva():
+    """We only test with positive imaginary part as the Humlicek approximation of the Faddeeva function is becoming
+    unstable in the negative imaginary part region.
+
+    That is also due to the nature of the Faddeeva function in that region.
+    """
+    size = int(1e3)
+    test_values1 = np.random.uniform(0, 0.1, size) + 1.0j * np.random.uniform(
+        0, 0.1, size
+    )
+    test_values2 = np.random.uniform(0, 1, size) + 1.0j * np.random.uniform(0, 1, size)
+    test_values3 = np.random.uniform(0, 10, size) + 1.0j * np.random.uniform(
+        0, 10, size
+    )
+    test_values4 = np.random.uniform(0, 100, size) + 1.0j * np.random.uniform(
+        0, 100, size
+    )
+    test_values5 = np.random.uniform(-0.1, 0, size) + 1.0j * np.random.uniform(
+        0, 1, size
+    )
+    test_values6 = np.random.uniform(-1, 0, size) + 1.0j * np.random.uniform(0, 1, size)
+    test_values7 = np.random.uniform(-10, 0, size) + 1.0j * np.random.uniform(
+        0, 10, size
+    )
+    test_values8 = np.random.uniform(-100, 0, size) + 1.0j * np.random.uniform(
+        0, 100, size
+    )
+    test_values = np.concatenate(
+        [
+            test_values1,
+            test_values2,
+            test_values3,
+            test_values4,
+            test_values5,
+            test_values6,
+            test_values7,
+            test_values8,
+        ]
+    )
+    assert np.allclose(
+        znp.faddeeva_humlicek(znp.asarray(test_values)),
+        wofz(test_values),
+        rtol=1e-06,
+        atol=1e-10,
+    )
+
+    test_values_tensor = znp.asarray(test_values)
+    with tf.GradientTape(persistent=True) as tape:
+        tape.watch(test_values_tensor)
+        result = znp.faddeeva_humlicek(test_values_tensor)
+        gradientstf = tape.gradient(result, test_values_tensor)
+    assert np.all(np.isfinite(gradientstf))
```

Comment:
@jonas-eschle what would be the proper assertion here to check the gradient.

Reply:
I don't really understand the equivalency here. The `z.math` functions take in a func with no arguments and zfit params. `zfit.Parameter` converts my complex input tensor into a float one and I want to check the gradients of `w(z)` with respect to `z` for a complex `z` tensor.

Reply:
We can test the derivatives of the `Voigt` with respect to `m`, `sigma` and `gamma` this way though

Reply:
> We can test the derivatives of the `Voigt` with respect to `m`, `sigma` and `gamma` this way though

yes, sorry, that's what I had in mind, I should have written it too. For x, one could use a numerical comparison, (like the numdifftools in `z.math`) or also just check for it to be finite, that should be the main possible issue

Reply:
I will maybe add something more here and let you know once it's ready to merge.